### PR TITLE
HPCC-16398 Force kill after sigterm attempt (exception for dali)

### PR DESCRIPTION
**Pull request by Michael-Gardner**
Signed-off-by: Michael Gardner michael.gardner@lexisnexis.com

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@mckellyln branch is up for review

File: initfiles/sbin/hpcc_setenv.in [hunk 1]
```diff
@@ -55,6 +70,10 @@ function pidwait_fn () {
 
     if [[ $TIMEOUT -le 0 ]] && ps -p $WATCH_PID -o command= | grep "${PROCESS}" &>/dev/null ; then
       log "Failed to kill ${PROCESS} with SIGTERM"
+      if [[ $FORCE -eq 1 ]]; then
+        kill -SIGKILL $WATCH_PID > /dev/null 2>&1
```
Comment (mckellyln) [inline initfiles/sbin/hpcc_setenv.in]: Add a log line here, as in -

log "killing ${PROCESS} (pid: $WATCH_PID) with SIGKILL"

**Review by mckellyln — APPROVED** [review]
These changes look good and help with making sure thor slaves end when asked.

File: initfiles/sbin/hpcc_setenv.in [hunk 2]
```diff
@@ -55,6 +70,11 @@ function pidwait_fn () {
...
+        log "killing ${PROCESS} (pid: $WATCH_PID) with SIGKILL"
```
Comment (mckellyln) [inline initfiles/sbin/hpcc_setenv.in]: Should we remove $PID file here before returning ?  As it does below -

rm -f $PID > /dev/null 2>&1

**Comment by mckellyln** [comment]
One more question about removing $PID file after kill -SIGKILL

File: initfiles/sbin/hpcc_setenv.in [hunk 3]
```diff
@@ -55,6 +70,12 @@ function pidwait_fn () {
...
+        rm -f $PID > /dev/null 2>&1
+        return 0
```
Comment (mckellyln) [inline initfiles/sbin/hpcc_setenv.in]: ok, good

**Review by mckellyln — APPROVED** [review]
Approved.

**Comment by mckellyln** [comment]
@richardkchapman @jakesmith ready for merge

**Pull request merged**